python_ta/transforms/type_inference_visitor.py

```diff
@@ -195,9 +195,27 @@ def visit_assign(self, node: astroid.Assign) -> None:
 
     def visit_annassign(self, node: astroid.AnnAssign) -> None:
         var_inf_type = self.lookup_typevar(node.target, node.target.name)
-        self.type_constraints.unify(var_inf_type, _node_to_type(node.annotation.name), node)
+        ann_type = self._ann_node_to_type(node.annotation)
+        self.type_constraints.unify(var_inf_type, ann_type, node)
         node.inf_type = NoType()
 
+    def _ann_node_to_type(self, node: astroid.Name) -> type:
+        ann_node_type = _node_to_type(node)
+        if isinstance(ann_node_type, GenericMeta) and ann_node_type.__args__ is None:
```

Comment:
This part should be changed. We tend so use both `Dict` and `dict` as aliases for `Dict[Any, Any]`, so this is how I'd like you to interpret all generic types with missing arguments.

Reply:
So I did initially have all the generic types set to `Any`, but then I saw #479 and thought that we were now inferring them as polymorphic? Otherwise we could have code such as
```
x: Dict

x = {'Hello', 0}
x = {1, 'Goodbye'}
```

Which would be permitted, even though `x` should be inferred as `Dict[str, int]` after the first assignment?